crates/e2e/src/setup/solver/mock.rs

```diff
@@ -15,32 +15,54 @@ use {
 /// A solver that does not implement any solving logic itself and instead simply
 /// forwards a single hardcoded solution.
 pub struct Mock {
-    /// The currently configured solution to return.
-    solution: Arc<Mutex<Option<Solution>>>,
+    /// Mock solver shareable state
+    state: State,
     /// Under which URL the solver is reachable by a driver.
     pub url: Url,
 }
 
+#[derive(Clone)]
+pub struct State {
+    /// In-memory set of the auctions received by the solver.
+    auctions: Arc<Mutex<Vec<Auction>>>,
```

Comment:
Since the mock solver should not do any test check, I decided to store the auctions, so the test case can fetch them to check data on it.